analyses/tumor-purity-exploration/01_explore-tumor-purity.Rmd

```diff
@@ -0,0 +1,503 @@
+  # find information to join in
+  select(sample_id, tumor_fraction) %>%
+  distinct() %>%
+  # Join tumor fractions with RNA metadata
+  right_join(same_extraction_stranded_metadata, by = "sample_id") %>%
+  # Keep only columns of interest, to avoid conflicts with future joins
+  select(contains("Kids_First"), sample_id, tumor_fraction, 
+         cancer_group, broad_histology) %>%
+  # remove any unknowns
+  drop_na(tumor_fraction)
+```
+
+The new `rna_stranded_df` contains information of interest for transcriptomic exploration: `tumor_fraction` values for stranded RNA-Seq samples that were from the same extraction as corresponding WGS for which the tumor purity was quantified.
+
+```{r}
+rna_stranded_df
+```
+
+Let's export this TSV for future use.
+
+```{r}
+write_tsv(rna_stranded_df, 
+          rna_stranded_df_file)
+```
+
+
+
+We might also like to see this distribution of values, since only these are relevant for RNA-level:
+
+```{r}
+summary(rna_stranded_df$tumor_fraction)
+
+ggplot(rna_stranded_df) + 
+  aes(x = tumor_fraction) + 
+  geom_histogram(color = "black", fill = "darkorchid2")
+```
+
+
+## Exploring thresholding
+
+
+If we were to potentially limit analyses to only tumors with a certain level of purity, how would that affect sample distributions? 
+In particular, how many samples would be left from each grouping if we removed at a certain threshold which might be, for example...
+
+- Overall unweighted median or mean tumor fraction
+- Weighted mean by group-size tumor fraction (but what is the grouping variable?)
+
+First let's join the new `rna_stranded_df` with the palette.
+
+```{r}
+rna_stranded_df <- rna_stranded_df %>%
+  inner_join(palette_mapping_df) %>%
+  select(Kids_First_Biospecimen_ID, 
+         tumor_fraction, 
+         cancer_group_display, 
+         cancer_group_hex, 
+         broad_histology_display,
+         broad_histology_order,
+         broad_histology_hex) %>%
+  drop_na() %>%
+  # reorder by median fraction and ensure "Other" is always last
+  mutate(cancer_group_display = forcats::fct_reorder(cancer_group_display, tumor_fraction), 
+         cancer_group_display = forcats::fct_relevel(cancer_group_display, "Other", after = Inf),
+         broad_histology_display = forcats::fct_reorder(broad_histology_display, tumor_fraction), 
+         broad_histology_display = forcats::fct_relevel(broad_histology_display, "Other", after = Inf))
+
+```
+
+Let's look at these threshold themselves:
+```{r}
+median_tumor <- median(rna_stranded_df$tumor_fraction, na.rm = TRUE)
+cat("median", median_tumor)
+
+unweighted_mean <- mean(rna_stranded_df$tumor_fraction, na.rm = TRUE)
+cat("\nunweighed mean", unweighted_mean)
+
+calc_weighted_mean <- function(df, group_var) {
+  df %>%
+    drop_na(tumor_fraction) %>%
+    add_count({{group_var}}) %>%
+    mutate(contribution = tumor_fraction * (n/sum(n))) %>%
+    summarize(weighted = sum(contribution)) %>% 
+    pull(weighted)
+}
```

Comment:
My strong preference is that all these functions interspersed throughout should get their own section at the top of the notebook and a little bit more documentation (as a treat).